src/agent/rustjail/src/container.rs

```diff
@@ -1022,7 +1020,7 @@ where
     })
 }
 
-fn do_exec(args: &[String]) -> Result<()> {
+fn do_exec(args: &[String]) -> ! {
```

Comment:
I thought the ! type wasn't in stable rust yet?

If it is available, then you can make do_init_child return Result<!> as well, which will simplify some of this.

Reply:
I think the never type is stable to used as function return types 
https://doc.rust-lang.org/reference/types/never.html

![image](https://user-images.githubusercontent.com/1013713/94527165-2aee1600-0269-11eb-8947-4ad84966c2bc.png)

Reply:
Result<!> can not be used for now

Reply:
Ah, right, I'd missed that ! was allowed just in that specific case.